File: 0792-binary-search/solution.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
    	int max = nums.size()-1;
    	int min = 0;
    	int guess;
        
	    while(max>=min){
	    	guess = (max+min)/2;
    		if(nums[guess] == target) return guess;
            if(nums[guess] < target) min = guess+1;
            else max = guess-1;
	    }
        return -1;
    }
};

File: 0792-binary-search/solution.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:

      minx = 0
      maxy = len(nums) - 1
      
      
      while minx <= maxy:
        var = int((minx+maxy)/2)
        
        if nums[var] == target:
          return var
        
        if nums[var] > target:
          maxy = var - 1
        
        if nums[var] < target:
          minx = var + 1
      
      return -1
